odoo/addons/base/models/ir_qweb.py

```diff
@@ -2211,8 +2387,54 @@ def _get_asset_nodes(self, bundle, css=True, js=True, debug=False, async_load=Fa
         else:
             return self._generate_asset_nodes_cache(bundle, css, js, debug, async_load, defer_load, lazy_load, media)
 
-    # other methods used for the asset bundles
+    # qweb cache feature
 
+    def _get_cache_key(self, cache_key):
+        """
+            Convert the template cache key item into a hashable key.
+            @params cache_key: tuple
+            @returns: tuple of hashable items
+        """
+        if not isinstance(cache_key, tuple):
+            raise ValueError("The cache key must be falsy or a python tuple.")
+        keys = []
+        datetimes = []
+        for item in cache_key:
+            try:
+                # use try catch instead of isinstance to detect lazy values
+                keys.append((item._name, tuple(item.ids)))
+                datetimes.extend(item.mapped('write_date'))
+            except AttributeError:
+                try:
+                    keys.append(float(item))
```

Comment:
To convert all numerics, because strings take up more space

Reply:
It depends of value, but why not (the size is more "stable"). Maybe add a comment because it is unclear.

Reply:
Actually, it is not correct. If you have 2 string a = "100" and the other b = "100.0" it will result the same keys, and it shouldn't because you want to keep it as a string.

Reply:
Before I had put an int before, but I told myself that it does not change anything. The format (location) of the int or float will be the same, no collision. It is only when reading the key (if one does) that there is a change.